blockchain/types/transaction_signing.go

```diff
@@ -248,6 +248,107 @@ type Signer interface {
+func NewLondonSigner(chainId *big.Int) Signer {
+	return londonSigner{eip2930Signer{NewEIP155Signer(chainId)}}
+}
+
+// ChainID returns the chain id.
+func (s londonSigner) ChainID() *big.Int {
+	return s.chainId
+}
+
+// Equal returns true if the given signer is the same as the receiver.
+func (s londonSigner) Equal(s2 Signer) bool {
+	x, ok := s2.(londonSigner)
+	return ok && x.chainId.Cmp(s.chainId) == 0
+}
+
+func (s londonSigner) Sender(tx *Transaction) (common.Address, error) {
+	if tx.Type() != TxTypeDynamicFee {
+		return s.eip2930Signer.Sender(tx)
+	}
+
+	if tx.ChainId().Cmp(s.chainId) != 0 {
+		return common.Address{}, ErrInvalidChainId
+	}
+
+	return tx.data.RecoverAddress(s.Hash(tx), true, func(v *big.Int) *big.Int {
+		// AL txs are defined to use 0 and 1 as their recovery
+		// id, add 27 to become equivalent to unprotected Homestead signatures.
+		V := new(big.Int).Add(v, big.NewInt(27))
+		return V
+	})
+}
+
+// SenderPubkey returns the public key derived from tx signature and txhash.
+func (s londonSigner) SenderPubkey(tx *Transaction) ([]*ecdsa.PublicKey, error) {
+	if tx.Type() != TxTypeDynamicFee {
+		return s.eip2930Signer.SenderPubkey(tx)
+	}
+
+	if tx.ChainId().Cmp(s.chainId) != 0 {
+		return nil, ErrInvalidChainId
+	}
+
+	return tx.data.RecoverPubkey(s.Hash(tx), true, func(v *big.Int) *big.Int {
+		// AL txs are defined to use 0 and 1 as their recovery
+		// id, add 27 to become equivalent to unprotected Homestead signatures.
+		V := new(big.Int).Add(v, big.NewInt(27))
+		return V
+	})
+}
+
+// SenderFeePayer returns the public key derived from tx signature and txhash.
+func (s londonSigner) SenderFeePayer(tx *Transaction) ([]*ecdsa.PublicKey, error) {
+	//EIP-1559(Dynamic fee transaction) tx don't supported fee-delegation.
+	return s.eip2930Signer.SenderFeePayer(tx)
+}
+
+// SignatureValues returns a new transaction with the given signature. This signature
+// needs to be in the [R || S || V] format where V is 0 or 1.
+func (s londonSigner) SignatureValues(tx *Transaction, sig []byte) (R, S, V *big.Int, err error) {
+	if tx.Type() != TxTypeDynamicFee {
+		return s.eip2930Signer.SignatureValues(tx, sig)
+	}
+
+	if len(sig) != crypto.SignatureLength {
+		panic(fmt.Sprintf("wrong size for signature: got %d, want %d", len(sig), crypto.SignatureLength))
+	}
+
+	if tx.data.ChainId().Sign() != 0 && tx.data.ChainId().Cmp(s.ChainID()) != 0 {
+		return nil, nil, nil, ErrInvalidChainId
+	}
+
+	R = new(big.Int).SetBytes(sig[:32])
+	S = new(big.Int).SetBytes(sig[32:64])
+	V = big.NewInt(int64(sig[crypto.RecoveryIDOffset]))
+
+	return R, S, V, nil
+}
+
+// Hash returns the hash to be signed by the sender.
+// It does not uniquely identify the transaction.
+func (s londonSigner) Hash(tx *Transaction) common.Hash {
+	if tx.Type() != TxTypeDynamicFee {
+		return s.eip2930Signer.Hash(tx)
+	}
+
+	infs := append([]interface{}{s.ChainID()}, tx.data.SerializeForSign()...)
+	return prefixedRlpHash(byte(tx.Type()), infs)
```

Comment:
```suggestion
	return prefixedRlpHash(byte(tx.Type()&0x0011), infs)
```
As we changed type from `0x1, 0x2` to `0x7801, 0x7802`
we need to do bit masking for this based on our SPEC.
```
// TxTypeAccessList (Klaytn Tx)
TxHashRLP = 0x01 || rlp([chainId, nonce, gasPrice, gasLimit, to, value, data, accessList, signature_y_parity, signature_r, signature_s])
TxHash = keccak256(TxHashRLP)

// TxTypeDynamicFee (Klaytn Tx)
TxHashRLP = 0x02 || rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, destination, amount, data, access_list, signature_y_parity, signature_r, signature_s]) 
TxHash = keccak256(TxHashRLP)
```

cc. @aidan-kwon @jimni1222